src/server/utils/devCdnFactory.js

```diff
@@ -0,0 +1,188 @@
+    module.legacyBrowser.url = module.legacyBrowser.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
+    module.node.url = module.node.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
+  });
+  return JSON.stringify(moduleMap, null, 2);
+};
+
+const matchPathToKnownRemoteModuleUrl = (
+  incomingRequestPath,
+  remoteModuleBaseUrls
+) => remoteModuleBaseUrls.find((remoteModuleBaseUrl) => {
+  const remoteModuleUrlOrigin = new URL(remoteModuleBaseUrl).origin;
+  const remoteModulePath = remoteModuleBaseUrl.replace(remoteModuleUrlOrigin, '');
+
+  return incomingRequestPath.startsWith(remoteModulePath);
+});
+
+export const oneAppDevCdnFactory = ({
+  localDevPublicPath,
+  remoteModuleMapUrl,
+  useLocalModules,
+  appPort,
+  useHost,
+  routePrefix = '',
+}) => {
+  if (!remoteModuleMapUrl && !useLocalModules) {
+    throw new Error('remoteModuleMapUrl is a required param when useLocalModules is not true');
+  }
+  if (!localDevPublicPath) { throw new Error('localDevPublicPath is a required param'); }
+  if (!appPort) { throw new Error('appPort is a required param'); }
+
+  if (remoteModuleMapUrl) {
+    console.log(`one-app-dev-cdn loading module map from ${remoteModuleMapUrl}`);
+  } else {
+    console.log('one-app-dev-cdn only using locally served modules');
+  }
+
+  const oneAppDevCdn = Fastify();
+
+  if (process.env.NODE_ENV === 'production') {
+    console.warn('do not include one-app-dev-cdn in production');
+    return oneAppDevCdn;
+  }
+  oneAppDevCdn.register(compress, {
+    global: false,
+  });
+  oneAppDevCdn.register(cors, {
+    origin: [
+      `http://localhost:${appPort}`,
+      `http://${ip.address()}:${appPort}`,
+      undefined,
+    ],
+  });
+
+  // for locally served modules
+  // oneAppDevCdn.use('/modules', express.static());
+  oneAppDevCdn.register(fastifyStatic, {
+    root: `${localDevPublicPath}/modules`,
+    prefix: `${routePrefix}/modules`,
+    index: false,
+
+  });
+
+  let remoteModuleBaseUrls = [];
+  // support one-app-cli's "serve-module"
+  // merge local with remote, with local taking preference
+  oneAppDevCdn.get(`${routePrefix}/module-map.json`, async (req, reply) => {
+    const hostAddress = useHost ? `http://${req.headers.host}` : `http://localhost:${process.env.HTTP_ONE_APP_DEV_CDN_PORT}`;
+
+    const remoteMapRequest = remoteModuleMapUrl
+      ? await got(remoteModuleMapUrl, {
+        agent: {
+          https: new ProxyAgent(),
+          http: new ProxyAgent(),
+        },
+      }).then(
+        (r) => {
+          // clear out remoteModuleBaseUrls as the new module map now has different urls in it
+          // not clearing would result in an ever growing array
+          remoteModuleBaseUrls = [];
+          const remoteModuleMap = JSON.parse(r.body);
+          const { modules } = remoteModuleMap;
+          const oneAppDevStaticsAddress = `${hostAddress}/static`;
+          Object.keys(modules).forEach((moduleName) => {
```

Comment:
`map` is a better fit than `forEach` since it will avoid mutation and generate a new array that can be assigned to `remoteModuleBaseUrls`